packages/pin-input/src/use-pin-input.ts

```diff
@@ -156,166 +170,173 @@ export function usePinInput(props: UsePinInputProps = {}) {
     [values, setValues, focusNext, onComplete, descendants.length],
   )
 
-  const clear = useCallback(() => {
+  const clear = React.useCallback(() => {
     const values: string[] = Array(descendants.length).fill("")
     setValues(values)
     const firstInput = descendants[0]
     firstInput.element?.focus()
   }, [descendants, setValues])
 
-  return {
-    id,
-    domContext,
-    setValue,
-    values,
-    setValues,
-    setMoveFocus,
-    clear,
-    onComplete,
-    placeholder,
-    manageFocus,
-    isDisabled,
-    isInvalid,
-  }
-}
+  const getNextValue = React.useCallback(
+    (value: string, eventValue: string) => {
+      let nextValue = eventValue
+      if (value?.length > 0) {
+        if (value[0] === eventValue.charAt(0)) {
+          nextValue = eventValue.charAt(1)
+        } else if (value[0] === eventValue.charAt(1)) {
+          nextValue = eventValue.charAt(0)
+        }
+      }
```

Comment:
Having trouble parsing this. What is this block doing? Handling a backspace input?